client/extensions/woocommerce/state/sites/settings/stripe-connect-account/actions.js

```diff
@@ -220,3 +224,124 @@ function deauthorizeFailure( siteId, action, errorMessage ) {
 		siteId,
 	};
 }
+
+/**
+ * Action Creator: Get the initial OAuth URL for connecting a Stripe Account.
+ *
+ * @param {Number} siteId The id of the site for which to create an account.
+ * @param {String} returnUrl The URL for Stripe to return the user to (to complete the setup)
+ * @param {String} [successAction=undefined] Optional action object to be dispatched upon success.
+ * @param {String} [failureAction=undefined] Optional action object to be dispatched upon error.
+ * @return {Object} Action object
+ */
+export const oauthInit = ( siteId, returnUrl, successAction = null, failureAction = null ) => (
+	dispatch,
+	getState
+) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	const initAction = {
+		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_OAUTH_INIT,
+		returnUrl,
+		siteId,
+	};
+
+	dispatch( initAction );
+
+	return request( siteId )
+		.post( 'connect/stripe/oauth/init', { returnUrl }, 'wc/v1' )
+		.then( data => {
+			dispatch( oauthInitSuccess( siteId, initAction, data ) );
+			if ( successAction ) {
+				dispatch( successAction( data ) );
+			}
+		} )
+		.catch( error => {
+			dispatch( oauthInitFailure( siteId, initAction, error ) );
+			if ( failureAction ) {
+				dispatch( failureAction( error ) );
+			}
+		} );
+};
+
+function oauthInitSuccess( siteId, { email }, { oauthUrl } ) {
```

Comment:
Even though there are some methods here like this that aren't exported, still think it would be nice to have some docblocks.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/19300/commits/3937df0266fb81e6a4bc2e4b44d35fde0ed8f572